thread-manager/src/tokio_runtime.rs

```diff
@@ -0,0 +1,185 @@
+use {
+    crate::{
+        policy::{apply_policy, parse_policy, CoreAllocation},
+        MAX_THREAD_NAME_CHARS,
+    },
+    serde::{Deserialize, Serialize},
+    solana_metrics::datapoint_info,
+    std::{
+        ops::Deref,
+        sync::{
+            atomic::{AtomicU64, AtomicUsize, Ordering},
+            Arc, Mutex,
+        },
+        time::Duration,
+    },
+    thread_priority::ThreadExt,
+};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(default)]
+pub struct TokioConfig {
+    ///number of worker threads tokio is allowed to spawn
+    pub worker_threads: usize,
+    ///max number of blocking threads tokio is allowed to spawn
+    pub max_blocking_threads: usize,
+    /// Priority in range 0..99
+    pub priority: u8,
+    pub policy: String,
+    pub stack_size_bytes: usize,
+    pub event_interval: u32,
+    pub core_allocation: CoreAllocation,
+}
+
+impl Default for TokioConfig {
+    fn default() -> Self {
+        Self {
+            core_allocation: CoreAllocation::OsDefault,
+            worker_threads: 8,
+            max_blocking_threads: 1,
+            priority: crate::policy::DEFAULT_PRIORITY,
+            policy: "OTHER".to_owned(),
+            stack_size_bytes: 2 * 1024 * 1024,
+            event_interval: 61,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct TokioRuntime {
+    pub tokio: tokio::runtime::Runtime,
+    pub config: TokioConfig,
+    pub counters: Arc<ThreadCounters>,
+}
+
+impl Deref for TokioRuntime {
+    type Target = tokio::runtime::Runtime;
+
+    fn deref(&self) -> &Self::Target {
+        &self.tokio
+    }
+}
+
+impl TokioRuntime {
+    /// Starts the metrics sampling task on the runtime to monitor how many workers are busy doing useful things.
```

Comment:
Not sure if the comment is properly formatted. I use "rewrap" plugin in VSCode (alt+q) to make comments fit.

Reply:
I have always thought cargo fmt should take care of these...

Reply:
me too, but it is not and so there is a plugin for that. weird